contracts/api3-server-v1/proxies/InverseApi3ReaderProxyV1.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import "./interfaces/IInverseApi3ReaderProxyV1.sol";
+
+/// @title An immutable proxy contract that inverts the value returned by an
+/// IApi3ReaderProxy data feed
+/// @dev This contract implements the AggregatorV2V3Interface to be compatible
+/// with Chainlink aggregators. This allows the contract to be used as a drop-in
+/// replacement for Chainlink aggregators in existing dApps.
+/// Refer to https://github.com/api3dao/migrate-from-chainlink-to-api3 for more
+/// information about the Chainlink interface implementation.
+contract InverseApi3ReaderProxyV1 is IInverseApi3ReaderProxyV1 {
+    /// @notice IApi3ReaderProxy contract address
+    address public immutable override proxy;
+
+    /// @param proxy_ IApi3ReaderProxy contract address
+    constructor(address proxy_) {
+        if (proxy_ == address(0)) {
+            revert ZeroProxyAddress();
+        }
+        proxy = proxy_;
+    }
+
+    /// @notice Returns the inverted value of the underlying IApi3ReaderProxy
+    /// @dev This inverts the 18-decimal fixed-point value using 1e36 / value.
+    /// It is assumed that operation might overflow if `baseValue` is too small.
+    /// @return value Inverted value of the underlying proxy
+    /// @return timestamp Timestamp from the underlying proxy
+    function read()
+        public
+        view
+        override
+        returns (int224 value, uint32 timestamp)
+    {
+        (int224 baseValue, uint32 baseTimestamp) = IApi3ReaderProxy(proxy)
+            .read();
+
+        if (baseValue == 0) {
```

Comment:
Is it worth to check this if we're not preventing the overflow on the next line?

Reply:
Yeah, to be honest I was quite hesitant to adding this check since a zero value for a data feed is unlikely and probably wrong

Reply:
I like the nice UX of getting and error back on cases when this happens and the caller also gets a refund of the unused gas but for our use case it might even make more sense to remove the check and wrap the math operation in `unchecked` to reduce gas even more. If we do this, we might consider documenting this so it doesn't get flagged in audits 🤷🏻